### PR TITLE
Merge VectorLeakyReLU for x86 and ARM

### PR DESCRIPTION
**Pull request by FedyuninV**

### CODE REVIEW

File: NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h [hunk 1]
```diff
@@ -944,6 +944,38 @@ inline void vectorHardSigmoid( const float* first, float* result, float slope, f
 	}
 }
 
+//------------------------------------------------------------------------------------------------------------
+
+static inline float32x4_t vectorLeakyReLUWorker( const float32x4_t& val,
+	const float32x4_t& alpha, const float32x4_t& zero )
+{
+	uint32x4_t upperMask = vcgeq_f32( val, zero );
+	float32x4_t lowerRes = vmulq_f32( alpha, val );
```
Comment (lordnn) [inline NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h]: `vmulq_n_f32()`

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h [hunk 2]
```diff
@@ -1202,6 +1202,30 @@ inline void vectorHardSigmoid( const float* first, float* result, float slope, f
...
+
+inline void vectorLeakyReLU( const float* first, float* result, float alpha, int vectorSize )
+{
+	const __m128 zeroSse = _mm_setzero_ps();
+	const __m128 alphaSse = _mm_set1_ps( alpha );
+
+	while( vectorSize >= 4 ) {
+		__m128 input = LoadSse4( first );
+		// result = x_pos + x_neg * alpha
+		StoreSse4( _mm_add_ps( _mm_max_ps( input, zeroSse ), _mm_mul_ps( _mm_min_ps( input, zeroSse ), alphaSse ) ),
```
Comment (lordnn) [inline NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h:1215]: ~~`StoreSse4( _mm_blendv_ps( _mm_mul_ps( input, alphaSse ), input, _mm_cmp_ps( input, zeroSse, _CMP_GE_OQ ) ) );`~~
AVX required

Comment (lordnn) [reply NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h:1215]: `StoreSse4( _mm_blendv_ps( _mm_mul_ps( input, alphaSse ), input, _mm_cmpge_ps( input, zeroSse ) ) );`
SSE4.1... Is it also too modern?

Comment (FedyuninV) [reply NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h:1215]: From the back of my head- I don't know. But figuring out why it was set to `-msse2` long time ago and is it still relevant nowadays...  Better  just leave it as it is and focus on AVX2+FMA or AVX-512 or other optimizations.

File: NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h (hunk 1 above)
Comment (FedyuninV) [reply NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h]: Fixed

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h [hunk 3]
```diff
@@ -1202,6 +1202,30 @@ inline void vectorHardSigmoid( const float* first, float* result, float slope, f
...
+			result );
+		first += 4;
+		result += 4;
+		vectorSize -= 4;
+	}
+
+	if( vectorSize > 0 ) {
+		__m128 input = LoadSse( first, vectorSize );
+		StoreSse( _mm_add_ps( _mm_max_ps( input, zeroSse ), _mm_mul_ps( _mm_min_ps( input, zeroSse ), alphaSse ) ),
```
Comment (lordnn) [inline NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h:1224]: `_mm_add_ps` may be changed to `_mm_or_ps`, IMHO

**Pull request merged**